homeassistant/components/wallbox/helpers.py

```diff
@@ -0,0 +1,59 @@
+"""Helpers for Wallbox integration."""
+
+from datetime import datetime as datetime_sys
+
+from dateutil import tz
+
+import homeassistant.util.dt as dt_util
+
+
+def map_schedule_item_to_ha(item: dict) -> dict:
+    """Map local schedule item to be more usable with HA."""
+    converted = {}
```

Comment:
why are we creating a new dict? we have a dict as input right?

Reply:
Same as above, not very experienced with Python (yet) :-)

In Python the input data is always mutable?

Reply:
a dict is, just change the attributes by setting them dict['attrib'] = new_value

Reply:
One more thing about this:
I'm creating a new dict since the service is not providing the full data received from the API.

I'm mapping the get_services output and set_services input as same 'model'. Currently API returns some extra data that is not required/accepted by the set schedules API.

Meaning that you get easily get the schedules from the service, modify time and pass the same or slightly modified data to the set schedules.

So, if I would re-use the provided dict (item) it would keep the obsolete data keys as well :-)

Reply:
just remove the key that has to go, item.pop('key_to_remove', None). I think you can do this in the coordinator function.